main.js

```diff
@@ -3,9 +3,38 @@ const EMPTY_HEART = '♡'
 const FULL_HEART = '♥'
 
 // Your JavaScript code goes here!
+const likeBtns = document.querySelectorAll("span.like-glyph")
+const errorBlock = document.querySelector("div#modal.hidden")
 
+const handleError = err => {
+  errorBlock.classList.remove("hidden")
+  errorBlock.innerText = err 
+  setTimeout(() => {
+    errorBlock.className = "hidden"
+  }, 3000)
+}
 
+const handleLikeBtn = event => {
+  if(event.target.innerText === EMPTY_HEART){
+    mimicServerCall()
+    .then(() => {
+      errorBlock.className = "hidden"
+      event.target.className = "activated-heart" 
+      event.target.innerText = FULL_HEART
+    })
+    .catch(handleError)
+  } else {
+    mimicServerCall()
+    .then(() => {
+      errorBlock.className = "hidden"
+      event.target.className = "like-glyph"
+      event.target.innerText = EMPTY_HEART
+    })
+    .catch(handleError)
+  }
```

Comment:
Can you think of a way to refactor this to have only one call to `mimicServerCall`?

Reply:
Would I make the server call, then do my if statement and run the then methods in there? So roughly something like 
:
`
mimicServerCall() 
 if(event.target.innerText === EMPTY_HEART){
    .then(() => {
      errorBlock.className = "hidden"
      event.target.className = "activated-heart" 
      event.target.innerText = FULL_HEART
    })
`

Reply:
Not quite. You can reduce the `.then`s and you can try moving the `if` statement around. If you have some extra time, come back and try a few things out.